### PR TITLE
Sonos binding: title channel for radio

### PR DESCRIPTION
**Pull request by ivivanov-bg**
- Update the currettitle and currenttrack channels with metadata from radio station even if opmlUrl is not set

Signed-off-by: Ivaylo Ivanov <ivivanov.bg@gmail.com>

### CODE REVIEW

File: extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/handler/ZonePlayerHandler.java [hunk 1]
```diff
@@ -769,20 +770,22 @@ protected void updateTrackMetaData() {
...
-                    artist = currentTrack.getCreator();
-                } else {
-                    artist = currentTrack.getAlbumArtist();
-                }
+                artist = !currentTrack.getAlbumArtist().isEmpty() ? currentTrack.getAlbumArtist()
+                        : currentTrack.getCreator();
 
                 if (!currentTrack.getTitle().contains("x-sonosapi-stream")) {
                     title = currentTrack.getTitle();
+                } else if (opmlUrl == null && currentUriMetaData != null) {
```
Comment (lolodomo) [inline extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/handler/ZonePlayerHandler.java]: This breaks the final else I just added that must be only if title contains x-sonosapi-stream.

I would propose:
```
                 if (!currentTrack.getTitle().contains("x-sonosapi-stream")) {
                     title = currentTrack.getTitle();
                 } else {
                     if (opmlUrl == null && currentUriMetaData != null) {
                         artist = currentUriMetaData.getTitle();
                         title = currentTrack.getStreamContent();
                     } else {
                         // For tune-in, don't reset the title as it was previously set in updateCurrentURIFormatted
                         // using the Opml HTTP query
                         title = stateMap.get("CurrentTitle");
                     }
                 }
```

Comment (ivivanov-bg) [inline extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/handler/ZonePlayerHandler.java]: what will be broken ?

The  branches are the exact same - you still have 3 options:
 1. Playing non-stream -> `currentTrack.getTitle().contains("x-sonosapi-stream")` will be `True`
 2. Playing a stream with `opmlUrl` not set -> then get the `artist` and `title` from `currentUriMetaData`
 3. Playing a stream with `opmlUrl` set -> use the `title` from the `stateMap`

In your case you just have two more brackets

Comment (lolodomo) [inline extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/handler/ZonePlayerHandler.java]: Yes, you're right, sorry for my wrong comment.

**Comment by lolodomo** [comment]
LGTM
@kaikreuzer : please merge so that I can take this PR into account and avoid a future conflict.

**Pull request merged**